core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -4080,6 +4080,46 @@ public static List reverse(List list) {
     return list;
   }
 
+  /** SQL {@code ARRAY_TO_STRING(array, delimiter)} function. */
+  public static String arrayToString(List list, String delimiter) {
+    return arrayToString(list, delimiter, null);
+  }
+
+  /** SQL {@code ARRAY_TO_STRING(array, delimiter, nullText)} function. */
+  public static String arrayToString(List list, String delimiter, @Nullable String nullText) {
+    StringBuilder sb = new StringBuilder();
+    int size = list.size();
+    for (int i = 0; i < size; i++) {
+      boolean shouldAddDelimiter = i != 0 && sb.length() > 0;
```

Comment:
Would it be enough to just use this?
```
boolean addDelimiter = sb.length() > 0;
```
In which case, we could remove the `size` and for `i`  and simply do:
```
for (Object item : list) {
```
Would that work?